core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -237,6 +235,22 @@ public InternalEngine(EngineConfig engineConfig) {
         logger.trace("created new InternalEngine");
     }
 
+    private LocalCheckpointTracker createLocalCheckpointTracker(
+        BiFunction<Long, Long, LocalCheckpointTracker> localCheckpointTrackerSupplier) throws IOException {
+        final long maxSeqNo;
+        final long localCheckpoint;
+        if (openMode == EngineConfig.OpenMode.CREATE_INDEX_AND_TRANSLOG) {
```

Comment:
paranoia - can you use a switch statement with a fallback logic and an exception on default?

Reply:
done in ea23a0d